src/libraries/Microsoft.PowerFx.Core/Entities/External/TableParameters.cs

```diff
@@ -0,0 +1,247 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.PowerFx.Core.Utils;
+using Microsoft.PowerFx.Types;
+
+namespace Microsoft.PowerFx.Core.Entities
+{
+    public class TableParameters
+    {
+        public SortRestrictions SortRestriction { get; init; }
+
+        public FilterRestrictions FilterRestriction { get; init; }
+
+        public SelectionRestrictions SelectionRestriction { get; init; }
+
+        public GroupRestrictions GroupRestriction { get; init; }
+
+        public IEnumerable<string> FilterFunctions { get; init; }
+
+        public IEnumerable<string> FilterSupportedFunctions { get; init; }
+
+        public PagingCapabilities PagingCapabilities { get; init; }
+
+        public IReadOnlyCollection<KeyValuePair<string, ColumnCapabilitiesBase>> ColumnsCapabilities { get; init; }
+
+        public bool SupportsDataverseOffline { get; init; }
+
+        public bool SupportsRecordPermission { get; init; }
+
+        public string TableName { get; init; }
+
+        public bool IsReadOnly { get; init; }
+
+        public FormulaType RecordType { get; init; }
```

Comment:
>     public FormulaType RecordType { get; init; } [](http://example.com/codeflow?start=3&length=49)

remove

Reply:
We can't as it's the lazy provider backing record type